src/pyhf/readxml.py

```diff
@@ -1,18 +1,28 @@
-from pyhf import schema
-from pyhf import compat
-from pyhf import exceptions
+from __future__ import annotations
 
 import logging
-
-from pathlib import Path
+import os
+import typing as T
 import xml.etree.ElementTree as ET
+from pathlib import Path
+
 import numpy as np
 import tqdm
 import uproot
 
+from pyhf import compat
+from pyhf import exceptions
+from pyhf import schema
+
 log = logging.getLogger(__name__)
 
+if T.TYPE_CHECKING:
+    PathOrStr = T.Union[str, os.PathLike[str]]
+else:
+    PathOrStr = T.Union[str, "os.PathLike[str]"]
```

Comment:
Uninformed question: As `typing.TYPE_CHECKING` is never run but only analyzed by `mypy` then there's no way to not take a hit here in coverage, correct?

![coverage_codecov](https://user-images.githubusercontent.com/5142394/184060456-322276b7-ccdd-4e78-a527-741c43f90abd.png)

Reply:
limitation on codecov's part. can't do much.

Reply:
@henryiii I assume that this is just something that we live with, correct?

Reply:
Never mind. PR #1937 fixes this with `.coveragerc`.

Reply:
What the if here? The string form is fine until you drop Pythons that don’t support it.